### ISSUE TITLE
Revisit whether CLI should wipe the emulator export directory

### ISSUE THREAD

**Issue by misha-reyzlin-hier** [firebase/firebase-tools#3600]
### [REQUIRED] Environment info

**firebase-tools:**

```
└ firebase --version
9.16.0
```

**Platform:**
macOS

### [REQUIRED] Test case

- Not sure what I should put here.

### [REQUIRED] Steps to reproduce
- Start the emulators with auth and database imports
- Shut down emulators with one ctrl+c 

### [REQUIRED] Expected behavior
Emulators shut down and data is exported.

### [REQUIRED] Actual behavior
Emulators print an error and the WHOLE FOLDER IS DELETED.

I am also unable to see the logs as the logs are deleted together with the working directory. 
```
i  emulators: Received SIGINT (Ctrl-C) for the first time. Starting a clean shutdown.
i  emulators: Please wait for a clean shutdown or send the SIGINT (Ctrl-C) signal again to stop right now.
i  Automatically exporting data using --export-on-exit "." please wait for the export to finish...
i  Found running emulator hub for project hierfoods-develop at http://localhost:4400
i  Exporting data to: /Users/misha/Source/hierfoods-quickfixes/functions
i  emulators: Received export request. Exporting data to /Users/misha/Source/hierfoods-quickfixes/functions.
⚠  emulators: Export failed: ENOENT: no such file or directory, link '/Users/misha/Source/hierfoods-quickfixes/functions/firebase-export-1627057765461MzTjpC' -> '/Users/misha/Source/hierfoods-quickfixes/functions'
⚠  Export request failed, see emulator logs for more information.
⚠  Automatic export to "." failed, going to exit now...
i  emulators: Shutting down emulators.
i  ui: Stopping Emulator UI
⚠  Emulator UI has exited upon receiving signal: SIGINT
i  functions: Stopping Functions Emulator
i  hosting: Stopping Hosting Emulator
i  database: Stopping Database Emulator
i  auth: Stopping Authentication Emulator
i  hub: Stopping emulator hub
i  logging: Stopping Logging Emulator
```

**Comment by misha-reyzlin-hier** [comment]
The way the emulators were started is probably to blame:

In the working path, we used:
```
firebase emulators:start --import . --export-on-exit .
```

This produced the relevant export files:
```
firebase-export-metadata.json
database_export
auth_export
``` 

And has worked OK for us until now, however at least only the exported files would be deleted by this bug and not the whole working folder alongside other important files. 

Still, loosing data on export is really bad. Also, perhaps the tool could verify that the path is not the same working directory if there is a risk of confusion at export time?

**Comment by Muritavo** [comment]
Almost lost a project today because of this.

Take a look at this PR, it proposes a basic fix for this cenario. https://github.com/firebase/firebase-tools/pull/4127

**Comment by yuchenshi** [comment]
My two cents: we should go with #4127 in the short term but find a way to do exports without wiping the target directory.

The wiping behavior was introduced in #3511 to solve an unrelated issue but maybe we can find a better solution. Ideas are welcomed!

**Comment by yuchenshi** [comment]
Workaround was released (#4127), repurpose this issue to track potentially revisiting the deletion behavior

**Comment by yuchenshi** [comment]
Consolidating the discussion to #3962.
